be/src/runtime/routine_load/data_consumer_pool.cpp

```diff
@@ -66,40 +70,70 @@ Status DataConsumerPool::get_consumer(StreamLoadContext* ctx, std::shared_ptr<Da
 }
 
 Status DataConsumerPool::get_consumer_grp(StreamLoadContext* ctx, std::shared_ptr<DataConsumerGroup>* ret) {
-    if (ctx->load_src_type != TLoadSourceType::KAFKA) {
-        return Status::InternalError("PAUSE: Currently only support consumer group for Kafka data source");
+    if (ctx->load_src_type != TLoadSourceType::KAFKA
+        && ctx->load_src_type != TLoadSourceType::PULSAR) {
+        return Status::InternalError("PAUSE: Currently only support consumer group for Kafka or Palsur data source");
     }
-    DCHECK(ctx->kafka_info);
-
-    // one data consumer group contains at least one data consumers.
-    int max_consumer_num = config::max_consumer_num_per_group;
-    size_t consumer_num = std::min((size_t)max_consumer_num, ctx->kafka_info->begin_offset.size());
-
-    std::shared_ptr<KafkaDataConsumerGroup> grp = std::make_shared<KafkaDataConsumerGroup>(consumer_num);
 
-    for (int i = 0; i < consumer_num; ++i) {
-        std::shared_ptr<DataConsumer> consumer;
-        RETURN_IF_ERROR(get_consumer(ctx, &consumer));
-        grp->add_consumer(consumer);
+    if (ctx->load_src_type == TLoadSourceType::KAFKA) {
+      DCHECK(ctx->kafka_info);
+  
+      // one data consumer group contains at least one data consumers.
+      int max_consumer_num = config::max_consumer_num_per_group;
+      size_t consumer_num = std::min((size_t)max_consumer_num, ctx->kafka_info->begin_offset.size());
+  
+      std::shared_ptr<KafkaDataConsumerGroup> grp = std::make_shared<KafkaDataConsumerGroup>(consumer_num);
+  
+      for (int i = 0; i < consumer_num; ++i) {
+          std::shared_ptr<DataConsumer> consumer;
+          RETURN_IF_ERROR(get_consumer(ctx, &consumer));
+          grp->add_consumer(consumer);
+      }
+
+      LOG(INFO) << "get consumer group " << grp->grp_id() << " with " << consumer_num << " consumers";
+      *ret = grp;
+      return Status::OK();
+    } else {
+      DCHECK(ctx->pulsar_info);
+      DCHECK_GE(ctx->pulsar_info->partitions.size(), 1);
+  
+      // Cumulative acknowledge is not supported for multiple topic subscribtion,
+      // so one consumer can only subscribe one topic/partition
+      int max_consumer_num = config::max_consumer_num_per_group;
```

Comment:
The default value of `max_consumer_num_per_group` is 3, which may be too small. To make the new config come into effect, the BE need to be restarted. So, how about using a specific config item to limit the consumer num of Pulsar ?

Reply:
Good point.